packages/react-core/src/components/DualListSelector/DualListSelectorListWrapper.tsx

```diff
@@ -0,0 +1,123 @@
+  displayOption?: (option: React.ReactNode) => boolean;
+}
+
+export const DualListSelectorListWrapperBase: React.FunctionComponent<DualListSelectorListWrapperProps> = ({
+  className,
+  children,
+  'aria-labelledby': ariaLabelledBy,
+  innerRef,
+  options = [],
+  selectedOptions = [],
+  onOptionSelect,
+  displayOption,
+  id = getUniqueId('dual-list-selector-list'),
+  ...props
+}: DualListSelectorListWrapperProps) => {
+  const [focusedOption, setFocusedOption] = React.useState('');
+  const menuRef = innerRef || React.useRef(null);
+  const { isTree } = React.useContext(DualListSelectorContext);
+
+  // sets up keyboard focus handling for the dual list selector menu child of the pane. This keyboard
+  // handling is applied whether the pane is dynamically built or passed via the children prop.
+  const handleKeys = (event: KeyboardEvent) => {
+    if (
+      !menuRef.current ||
+      (menuRef.current !== (event.target as HTMLElement).closest('.pf-c-dual-list-selector__menu') &&
+        !Array.from(menuRef.current.getElementsByClassName('pf-c-dual-list-selector__menu')).includes(
+          (event.target as HTMLElement).closest('.pf-c-dual-list-selector__menu')
+        ))
+    ) {
+      return;
+    }
+    event.stopImmediatePropagation();
+    const validOptions = isTree
+      ? (Array.from(
+          menuRef.current.querySelectorAll(
+            '.pf-c-dual-list-selector__item-toggle, .pf-c-dual-list-selector__item-check > input'
+          )
+        ) as Element[])
+      : (Array.from(menuRef.current.getElementsByTagName('LI')) as Element[]).filter(
+          el => !el.classList.contains('pf-m-disabled')
+        );
+    const activeElement = document.activeElement;
+    handleArrows(
+      event,
+      validOptions,
+      (element: Element) => activeElement.contains(element),
+      (element: Element) => {
+        if (element.classList.contains('.pf-c-dual-list-selector__list-item')) {
+          setFocusedOption(element.id);
+        } else {
+          setFocusedOption(element.closest('.pf-c-dual-list-selector__list-item').id);
+        }
+        return element;
+      },
+      ['.pf-c-dual-list-selector__item-toggle', '.pf-c-dual-list-selector__item-check > input'],
+      undefined,
+      false,
+      false,
+      false
+    );
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('keydown', handleKeys);
+    return () => {
+      window.removeEventListener('keydown', handleKeys);
+    };
+  }, [menuRef.current]);
+
+  return (
+    <div className={css(styles.dualListSelectorMenu, className)} ref={menuRef} tabIndex={0} {...props}>
+      <DualListSelectorListContext.Provider
+        value={{
+          setFocusedOption,
+          isTree,
+          focusedOption,
+          ariaLabelledBy,
+          displayOption,
+          selectedOptions,
+          id,
+          options,
+          onOptionSelect
+        }}
+      >
+        {children ? children : <DualListSelectorList />}
```

Comment:
So if there are no `children` render empty `DualListSelectorList`?  What happens if we don't render anything.

Reply:
The list should have children in the case of the composable dual list selector.
The list uses it's options prop (passed via context) if it's a dynamically built dual list selector.
if there are no children and no options prop, it just renders an empty list.